src/components/User/UserBuzz.tsx

```diff
@@ -20,20 +20,33 @@ export function UserBuzz({
   ...textProps
 }: Props) {
   if (!user) return null;
+  const { balance } = user;
 
   const content = (
     <Text color="accent.5" transform="uppercase" {...textProps}>
       <Group spacing={4} noWrap>
         <IconBolt size={iconSize} color="currentColor" fill="currentColor" />
-        <Text size={textSize} weight={600} lh={1.2} sx={{ fontVariantNumeric: 'tabular-nums' }}>
-          {withAbbreviation ? abbreviateNumber(user.balance) : user.balance.toLocaleString()}
+        <Text size={textSize} weight={600} lh={0} sx={{ fontVariantNumeric: 'tabular-nums' }}>
+          {balance === null ? (
+            <Loader size="sm" variant="dots" color="accent.5" />
+          ) : withAbbreviation ? (
+            abbreviateNumber(balance)
+          ) : (
+            balance.toLocaleString()
+          )}
         </Text>
       </Group>
     </Text>
   );
 
   return withTooltip ? (
-    <Tooltip label={`Total balance: ${user.balance.toLocaleString()}`}>{content}</Tooltip>
+    <Tooltip
+      label={`Total balance: ${
+        balance === null ? <Loader size="sm" variant="dots" /> : balance.toLocaleString()
+      }`}
+    >
+      {content}
+    </Tooltip>
```

Comment:
I'm curious how the loader would look in the tooltip? To me it would feel better to have a literal "Loading..." or "Calculating..." which would make it more explicit that something is going on in case users hover over
```suggestion
    <Tooltip
      label={`Total balance: ${
        balance === null ? 'Calculating...' : balance.toLocaleString()
      }`}
    >
      {content}
    </Tooltip>
```